mizu-core/src/cpu.rs

```diff
@@ -917,16 +917,16 @@ impl Cpu {
                     HaltMode::HaltNoRunInterrupt
                 };
 
-                0
+                Ok(0)
             }
             Opcode::Stop => {
                 // TODO: respect wait time for speed switch
                 bus.enter_stop_mode();
-                0
+                Ok(0)
             }
-            Opcode::Illegal => todo!(),
-            Opcode::Prefix => unreachable!(),
-        };
+            Opcode::Illegal => Err("undecodable opcode"),
+            Opcode::Prefix => Err("unreachable"),
```

Comment:
This is literally unreachable, so u can keep it as `unreachable!()`, as its just used during decoding